composer/checkpoint/state_dict.py

```diff
@@ -158,6 +169,82 @@ def _get_model_state_dict_with_fsdp_context_manager(model: nn.Module, sharded_st
     return model_state_dict
 
 
+def get_resumption_state_dict(state: State) -> Dict[str, Any]:
+    """Generate the state dict for any objects needed for resumption.
+
+    This includes:
+        * timestamp
+        * scheduler
+        * dataset_state
+        * scaler
+        * rank_zero_seed
+        * callbacks
+        * algorithms
```

Comment:
iirc this list is stored as a constant in `state.py` as serialized atrributes. Is this the same / if so can we use that to avoid changing one part and ignoring another?

We can then test to ensure every attr in the list is included in resumption state dict to avoid discrepancies if someone changes it

Reply:
No. I think this is way more readable to explicitly put each item

Reply:
We will remove that list from state.py once we switch to this API